File: app/src/main/java/com/kang/resume/preview/fragment/BaseFragment.java
package com.kang.resume.preview.fragment;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.kang.resume.BuildConfig;

/**
 * 类描述：基础fragment
 */
public abstract class BaseFragment extends Fragment {

    protected Activity activity;
    private View view;


    protected abstract int initLayout(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState);

    protected abstract void initView(View view);

    protected abstract void init(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState);

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        if (view == null) {
            view = View.inflate(activity, initLayout(inflater, container, savedInstanceState), null);
            initView(view);


            if(BuildConfig.DEBUG){
                init(inflater, container, savedInstanceState);
            }else {
                try {
                    init(inflater, container, savedInstanceState);
                } catch (Exception e) {

                }
            }
        } else {
            ViewGroup parent = (ViewGroup) view.getParent();
            if (parent != null) {
                parent.removeView(view);
            }
        }
        return view;
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        this.activity = (Activity) context;
    }
}

File: app/src/main/java/com/kang/resume/base/BaseFragment.kt
package com.kang.resume.base

import android.app.Activity
import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.annotation.LayoutRes
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModel
import com.kang.resume.R
import com.kang.resume.pro.IView
import com.lxj.xpopup.XPopup
import com.lxj.xpopup.impl.LoadingPopupView
import kotlinx.coroutines.DelicateCoroutinesApi


/**
 * 类描述：
 * author:kanghuicong
 */
abstract class BaseFragment<V : ViewDataBinding, VM : BaseViewModel> : Fragment(),IView {

    lateinit var mBinding: V

    lateinit  var mVm: VM

    lateinit var activity: AppCompatActivity


    @LayoutRes
    abstract fun initLayout(): Int

    abstract fun initViewCreated(view: View)

    abstract fun initViewModel(): VM

    @DelicateCoroutinesApi
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        mBinding = DataBindingUtil.inflate(inflater, initLayout(), container, false)
        mVm = initViewModel()
        //初始化弹窗
        initPopup(activity,mVm,this)

        mBinding.lifecycleOwner = this
        return mBinding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initViewCreated(view)
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        this.activity = context as AppCompatActivity
    }

    override fun onDestroy() {
        super.onDestroy()
        mBinding.unbind()
    }
}